mi/reporting/resources.py

```diff
@@ -0,0 +1,73 @@
+from datetime import date, datetime
+from pathlib import Path
+from typing import Generator
+
+from helpers.log import log
+from mi.reporting.paths import PATH_TO_QUERIES
+
+RESOURCE_PREFIX = "nhsd-nrlf"
+LAMBDA_NAME = RESOURCE_PREFIX + "--{workspace}--mi--sql_query"
+SECRET_NAME = RESOURCE_PREFIX + "--{workspace}--{workspace}--{operation}_password"
+DB_CLUSTER_NAME = RESOURCE_PREFIX + "-{env}-aurora-cluster"
+
+
+@log("Got credentials")
+def get_credentials(session, workspace: str, operation: str = "read") -> dict:
+    secret_name = SECRET_NAME.format(workspace=workspace, operation=operation)
+    client = session.client("secretsmanager")
+    response = client.get_secret_value(SecretId=secret_name)
+    return {"user": f"{workspace}-{operation}", "password": response["SecretString"]}
+
+
+@log("Got SQL statement {__result__}")
+def each_sql_statement() -> Generator[tuple[str, str], None, None]:
```

Comment:
Unless we plan on extending this to other SQL actions we could also include the word report in here `each_report_sql_statement`